lms/djangoapps/courseware/toggles.py

```diff
@@ -126,14 +127,20 @@
     WAFFLE_FLAG_NAMESPACE, 'mfe_proctored_exams', __name__
 )
 
+PREVIEW_BASE = settings.FEATURES.get('PREVIEW_LMS_BASE', None)
+
 
 def mfe_special_exams_is_active(course_key: CourseKey) -> bool:
     """
     Can we see a course special exams in the Learning MFE?
     """
+    hostname = get_current_request_hostname()
     # DENY: Old Mongo courses don't work in the MFE.
     if course_key.deprecated:
         return False
+    # DENY: Course preview doesn't work in the MFE
+    if PREVIEW_BASE and hostname == PREVIEW_BASE:
+        return False
```

Comment:
To keep the "am I in preview mode?" logic consistent across the LMS, I recommend just using the [`in_preview_mode`](https://github.com/edx/edx-platform/blob/2a9b1a495e829091edbb99d35cd73b96cc942134/lms/djangoapps/courseware/access_utils.py#L96-L102) function. It handles the special cases you and @doctoryes discussed, as well as the case where the hostname includes a port number.